convey/conveymetric/metric.go

```diff
@@ -0,0 +1,76 @@
+package conveymetric
+
+import (
+	"fmt"
+	"github.com/Comcast/webpa-common/convey"
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/provider"
+	"regexp"
+	"strings"
+)
+
+const UnknownLabel = "unknown"
+
+type MetricClosure func()
+
+type CMetric interface {
+	Update(data convey.C) (MetricClosure, error)
+}
+
+var (
+	validKeyRegex   = regexp.MustCompile(`^[a-zA-Z_:]+$`)
+	replaceRegex    = regexp.MustCompile(`[\W]+`)
+	underscoreRegex = regexp.MustCompile(`_+`)
+)
+
+func NewConveyMetric(provider provider.Provider, tag string, name string) CMetric {
```

Comment:
It's best not to depend on the `Provider` interface.  Instead, simply pass a `metrics.Gauge`.  Rely on enclosing application code to configure the provider and create the gauge.  This makes this much more testable, and isolates this code from creational semantics.  Creating a gauge is a nontrivial thing that involves labels, naming conventions, etc.  That's best left outside of library code.